inference-engine/src/plaidml_plugin/ops/shuffle_channels.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "ngraph/opsets/opset.hpp"
+#include "ngraph/opsets/opset3.hpp"
+#include "plaidml/op/op.h"
+#include "plaidml_ops.hpp"
+
+using namespace plaidml;          // NOLINT[build/namespaces]
+using namespace InferenceEngine;  // NOLINT[build/namespaces]
+
+namespace PlaidMLPlugin {
+
+static OpRegistration reg("shuffleChannels", [](const Context& ctx) {
+  auto* layer = ngraph::as_type<ngraph::opset3::ShuffleChannels>(ctx.layer);
+  IE_ASSERT(ctx.operands.size() == 1);
+  auto I = ctx.operands.at(0);
+  auto group = layer->get_group();
+  //  the axis parameter doesn't show in the formula.
+  //  not sure why we need axis, so just leave it alone for now.
+
+  std::vector<edsl::TensorDim> original_dims(I.rank());
+  I.bind_dims(original_dims);
+  // the channel dim size have to be dividable by group.
+  auto channel_dim = original_dims[1];
+
+  // follow the openvino op spec
+  // x' = reshape(x, [N, group, C / group, H * W])
```

Comment:
I think the spec given for `axis = 1` is equivalent to:
```
x' = reshape(x, [N, group, C / group, H, W])
x'' = transpose(x', [0, 2, 1, 3, 4])
y = reshape(x'', [N, C, H, W])
```
I think for other axis values, they intend a similar approach but with C in a different position. E.g. for `axis = 3` I think the intent is
```
x' = reshape(x, [N, H, W, group, C / group])
x'' = transpose(x', [0, 1, 2, 4, 3])
y = reshape(x'', [N, H, W, C])
```
I think in general, the algorithm is along the lines of the following pseudocode:
```
channel_group_dims = original_dims;
channel_group_dims[axis] = channel_group_dims[axis] / group;
channel_group_dims.insert(axis, group);
reshape_I = reshape(I, channel_group_dims);

vector<integer> order = range(reshape_I.rank());
swap(order[axis], order[axis+1]);
transpose_I = transpose(reshape_I, order);

O = reshape(transpose_I, original_dims);
```

Reply:
very thanks for your comment. it's really helpful. and since axis matter here according to  openvino spec, the rang of axis is [-4, 3], so I also make some pre-process for axis here.